vitals/LiIonBattery.java

```diff
@@ -0,0 +1,113 @@
+package vitals;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.function.Predicate;
+
+/**
+ * @author {@literal Jayaram Naveenkumar (jayaram.naveenkumar@in.bosch.com)}
+ */
+public class LiIonBattery {
+
+    private Map<BMS, Float> input;
+
+    public LiIonBattery(
+        float temperature,
+        float soc,
+        float chargeRate
+    )
+    {
+        saveInput(temperature, soc, chargeRate);
+    }
+
+    private void saveInput(
+        float temperature,
+        float soc,
+        float chargeRate
+    )
+    {
+        input = new TreeMap<>();
+        input.put(BMS.TEMPERATURE, temperature);
+        input.put(BMS.SOC, soc);
+        input.put(BMS.CHARGINGRATE, chargeRate);
+    }
+
+    public static final class Temperature {
+
+        private static final float MAX_TEMPERATURE_THRESHOLD = 45;
+        private static final float MIN_TEMPERATURE_THRESHOLD = 0;
+
+        private Temperature() {
+        }
+
+        public static Predicate<Float> isBeyondThreshold() {
+            return soc -> soc > MAX_TEMPERATURE_THRESHOLD;
+        }
+
+        public static Predicate<Float> isBelowThreshold() {
```

Comment:
Duplicate Code among isBelow*** and isAbove**** methods... good thought to use High Order functions ...why don't you create common reusable  functionality (one for belowThreshold check  and another for above threadshold check)

Reply:
yes Refactored